Havtorn/Source/Engine/Graphics/RenderManager.cpp

```diff
@@ -1271,15 +1478,62 @@ namespace Havtorn
 		auto textureBank = GEngine::GetTextureBank();
 		for (U8 drawCallIndex = 0; drawCallIndex < static_cast<U8>(staticMeshComp->DrawCallData.size()); drawCallIndex++)
 		{
-			// Load Textures
+			SEngineGraphicsMaterial materialCopy = materialComp->Materials[drawCallIndex];
 			std::vector<ID3D11ShaderResourceView*> resourceViewPointers;
-			resourceViewPointers.resize(TexturesPerMaterial);
-			for (U8 textureIndex = 0, pointerTracker = 0; textureIndex < TexturesPerMaterial; textureIndex++, pointerTracker++)
+			
+			std::map<U16, U16> textureIndices;
+			auto findTextureByIndex = [&] (SEngineGraphicsMaterialProperty& materialProperty)
```

Comment:
[&] (SEngineGraphicsMaterialProperty& materialProperty), please remove white space between [&] and (SEn...)